src/insertion.jl

```diff
@@ -30,6 +30,14 @@ function.
 """
 isinsertable(::AbstractIndices) = false
 
+function safe_convert(::Type{I}, i) where {I}
```

Comment:
One more thing: adding a `safe_convert(::Type{I}, i::I) where {I} = i` method here should speed this up significantly.

Reply:
Sorry for the "slight" delay. Just implemented your suggestion